test/models/api_namespace_test.rb

```diff
@@ -1,7 +1,37 @@
 require "test_helper"
 
 class ApiNamespaceTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  setup do
+    @subdomain = subdomains(:public)
+    @subdomain.update(api_plugin_events_enabled: true)
+    @subdomain_events_api = api_namespaces(:plugin_subdomain_events)
+    @subdomain_events_api.api_resources.destroy_all
+    @message_thread = message_threads(:public)
+    @message = @message_thread.messages.create!(content: "Hello")
+    Sidekiq::Testing.fake!
+  end
+
+  test "plugin: subdomain/subdomain_events -> tracks message creation by creating ApiResource" do
+    service = SubdomainEventsService.new(@message)
+    assert_difference "ApiResource.count", +1 do      
+      assert_changes "@subdomain_events_api.create_api_actions.first.lifecycle_stage" do        
+        service.track_event
+        Sidekiq::Worker.drain_all
+      end
+    end
+    @subdomain_events_api.api_resources.reload.last.properties
+  end
+
+  test "plugin: subdomain/subdomain_events -> tracks message creation by creating ApiResource & running actions" do
+    skip('investigate why api_resource_id: is not set on the @subdomain_events_api.create_api_actions.first')
```

Comment:
@Pralish any idea what's going on here?

Reply:
> @Pralish any idea what's going on here?

@donrestarone , Just checked the issue. 

It should be @subdomain_events_api.executed_api_actions.first 
OR
 if you want to be more specific
@subdomain_events_api.executed_api_actions.where(type: 'CreateApiAction').first
OR
@subdomain_events_api.api_resources.last.create_api_actions.first

@subdomain_events_api.create_api_actions.first is the template that api_resources will use to create the create_api_actions

Let me know if that makes sense